File: tests/fixtures/sample_c/sample_module.c
#include <stdio.h>
#include <stdlib.h>
#include "utils.h"

typedef struct {
    int x;
    int y;
} Point;

typedef enum {
    RED,
    GREEN,
    BLUE,
} Color;

typedef int (*BinaryOp)(int, int);

int add(int a, int b) {
    return a + b;
}

int multiply(int a, int b) {
    return a * b;
}

int compute(int a, int b) {
    int sum = add(a, b);
    int product = multiply(a, b);
    return sum + product;
}

void print_point(Point p) {
    printf("(%d, %d)\n", p.x, p.y);
}

File: tests/fixtures/sample_zig/sample_module.zig
const std = @import("std");
const math = @import("math.zig");

pub const MAX_SIZE: usize = 1024;

const Color = enum { Red, Green, Blue };

const Point = struct {
    x: f64,
    y: f64,
};

const Shape = union {
    circle: f64,
    rect: Point,
};

pub fn add(a: i32, b: i32) i32 {
    return a + b;
}

pub fn multiply(a: i32, b: i32) i32 {
    return a * b;
}

fn compute(a: i32, b: i32) i32 {
    const sum = add(a, b);
    const product = multiply(a, b);
    return sum + product;
}
